alacritty/src/input.rs

```diff
@@ -768,6 +776,19 @@ impl<'a, T: EventListener, A: ActionContext<T>> Processor<'a, T, A> {
 
                 // Don't suppress when there has been a `ReceiveChar` action
                 *suppress_chars.get_or_insert(true) &= binding.action != Action::ReceiveChar;
+
+                // Hide the cursor if necessary
+                if self.ctx.config().ui_config.mouse.hide_when_typing {
+                    let hide_cursor = match binding.action {
+                        Action::Esc(_) => true,
+                        Action::ViMotion(_) => true,
+                        _ => false
+                    };
+
+                    if hide_cursor {
+                        self.ctx.window_mut().set_mouse_visible(false);
+                    }
+                }
```

Comment:
Why does this need to be done here instead of the action handlers?

Reply:
Good point, moved